### PR TITLE
Add support for Equinox XL

### PR DESCRIPTION
**Pull request by ai03-2725**
## Description

Adds support for the Equinox XL keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds support for Equinox XL

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ai03/equinox_xl/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS
+	),
+
+	[3] = LAYOUT(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS
+	),
```
Comment (dunk2k) [inline keyboards/ai03/equinox_xl/keymaps/default/keymap.c:33]: non-functional layers
```suggestion
```

File: keyboards/ai03/equinox_xl/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS
+	),
+
+	[3] = LAYOUT(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS
+	),
```
Comment (dunk2k) [inline keyboards/ai03/equinox_xl/keymaps/via/keymap.c:33]: remove non-functional layers
```suggestion
```

File: keyboards/ai03/equinox_xl/keymaps/via/rules.mk [hunk 3]
```diff
@@ -0,0 +1 @@
+VIA_ENABLE = yes
```
Comment (dunk2k) [inline keyboards/ai03/equinox_xl/keymaps/via/rules.mk:1]:
```suggestion
VIA_ENABLE = yes

```

File: keyboards/ai03/equinox_xl/info.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        "rows": ["B7", "F4", "F6", "F7"]
+    },
+    "processor": "atmega32u4",
+    "url": "https://ai03.com/projects/equinox-xl/",
+    "usb": {
+        "device_version": "0.0.1",
+        "pid": "0x0020",
+        "vid": "0xA103"
+    },
+    "layouts": {
```
Comment (dunk2k) [inline keyboards/ai03/equinox_xl/info.json:33]: could matrix co-ordinates (e.g. `1,12`) for the following keys be confirmed?:
- ISO Hash
- ISO Enter
- 2.25u Left Shift
- 2.75u Right Shift
- 7u Spacebar

File: keyboards/ai03/equinox_xl/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT(
+		KC_ESC,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC,
+		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+		KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_GRV,
```
Comment (dunk2k) [inline keyboards/ai03/equinox_xl/keymaps/default/keymap.c:10]:
```suggestion
		KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_GRV,
```

File: keyboards/ai03/equinox_xl/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT(
+		KC_ESC,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC,
+		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+		KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_GRV,
```
Comment (dunk2k) [inline keyboards/ai03/equinox_xl/keymaps/via/keymap.c:10]:
```suggestion
		KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_GRV,
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/ai03/equinox_xl/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
```
Comment (waffle87) [inline keyboards/ai03/equinox_xl/keymaps/default/keymap.c:1]:
```suggestion
// Copyright 2024 ai03
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/ai03/equinox_xl/keymaps/default/readme.md
Comment (waffle87) [inline keyboards/ai03/equinox_xl/keymaps/default/readme.md:1]: This file should be removed.

File: keyboards/ai03/equinox_xl/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
```
Comment (waffle87) [inline keyboards/ai03/equinox_xl/keymaps/via/keymap.c:1]:
```suggestion
// Copyright 2024 ai03
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/ai03/equinox_xl/keymaps/via/readme.md
Comment (waffle87) [inline keyboards/ai03/equinox_xl/keymaps/via/readme.md:1]: This file should be removed.

File: keyboards/ai03/equinox_xl/info.json (hunk 4 above)
Comment (ai03-2725) [reply keyboards/ai03/equinox_xl/info.json:33]: I apologize for the delay; ended up busy with things that came up in life.

I've double-checked the requested keys:
- ISO hash - common matrix position with ANSI pipe (0, 13)
- ISO enter - common matrix position with ANSI enter (1, 13)
- 2.25U left shift - common matrix position with 1.25U left shift (2, 0)
- 2.75U right shift - common matrix position with the 1U to the right of 1.75U right shift (2, 13)
- 7U spacebar - common matrix position with the 1U central key of split space (3, 7)

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**